ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -627,10 +627,14 @@ func (r *RayClusterReconciler) updateStatus(instance *rayiov1alpha1.RayCluster)
 	}
 	// only in invalid status that we update the status to unhealthy.
 	if !isValid {
-		instance.Status.State = rayiov1alpha1.Unhealthy
+		if updateErr := r.updateClusterState(instance, rayiov1alpha1.Unhealthy); updateErr != nil {
+			log.Error(updateErr, "RayCluster update state error", "cluster name", instance.Name)
+		}
 	} else {
 		if utils.CheckAllPodsRunnning(runtimePods) {
-			instance.Status.State = rayiov1alpha1.Ready
+			if updateErr := r.updateClusterState(instance, rayiov1alpha1.Ready); updateErr != nil {
```

Comment:
do we want to update the state immediately? Can we update the status at the end of the reconcile loop?

I think it's empty because of `if .. else` logic? If `utils.CheckAllPodsRunnning(runtimePods) == false`, then nothing assigned to `instance.Status.State`?

Reply:
/cc @scarlet25151

Reply:
@Jeffwan sorry, my bad. I didn't notice that we've updated the status at line 643. I used the v0.2.0 image so the state is always empty.

Reply:
I just fixed by using nightly version of kuberay